modules/phantompeaktools_runSpp.cfmod.pl

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env perl
+use warnings;
+use strict;
+use Getopt::Long;
+use FindBin qw($Bin);
+use lib "$FindBin::Bin/../source";
+use CF::Constants;
+use CF::Helpers;
+use File::Copy qw(move);
+
+##########################################################################
+# Copyright 2014, Philip Ewels (phil.ewels@babraham.ac.uk)               #
+#                                                                        #
+# This file is part of Cluster Flow.                                     #
+#                                                                        #
+# Cluster Flow is free software: you can redistribute it and/or modify   #
+# it under the terms of the GNU General Public License as published by   #
+# the Free Software Foundation, either version 3 of the License, or      #
+# (at your option) any later version.                                    #
+#                                                                        #
+# Cluster Flow is distributed in the hope that it will be useful,        #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of         #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
+# GNU General Public License for more details.                           #
+#                                                                        #
+# You should have received a copy of the GNU General Public License      #
+# along with Cluster Flow.  If not, see <http://www.gnu.org/licenses/>.  #
+##########################################################################
+
+my %requirements = (
+	'cores' 	=> '1',
+	'memory' 	=> '2G',
+	'modules' 	=> 'samtools',
+	'time' 		=> sub {
+
+		my $cf = $_[0];
+		my $num_files = $cf->{'num_starting_merged_aligned_files'};
+		# Default to 1 if none were parsed for whatever reason
+		$num_files = ($num_files > 0) ? $num_files : 1;
+		# Picard usually takes around 30 minutes per BAM file?
+		return CF::Helpers::minutes_to_timestamp ($num_files * 1 * 60);
+	}
+);
+
+# The help text
+my $helptext = "".("-"x15)."\nPhantompeaktools run SPP".("-"x15)."\n
+Takes (deduplicated) bam files as input and runs cross correlation analysis, using the run_spp.r script from the pantompeaktools packeage. Outputs for each bam file are basename_crosscorrelation.pdf (a cross correlation plot) and  basename_crosscorrelation.txt (a text file with various cross correlation statistics). For convenience the input bam file is also passed on as output to be used by the next script in the pipeline.\n\n";
+
+# The script we will run. Now hard coded path. FIX THIS!!
```

Comment:
This module could be written in R directly, which _may_ make life easier. See the [example R module](https://github.com/ewels/clusterflow/blob/master/modules/example_module.R).

I started writing an R module a while back but got really frustrated with the tool so abandoned it. This could be a good pilot module though! Quite a bit of faff+ time to translate it though...